hub/codec/codecs.py

```diff
@@ -0,0 +1,380 @@
+from io import BytesIO
+import numcodecs
+from numcodecs.abc import Codec
+
+import numpy as np
+from PIL import Image
+
+
+class Base:
+    """Base class for numcodec compressors"""
+
+    def encode(self, array: np.ndarray) -> bytes:
+        raise NotImplementedError()
+
+    def decode(self, bytes: bytes) -> np.ndarray:
+        raise NotImplementedError()
+
```

Comment:
we need an @property on the base class (& all implementations) that denote if the compression operates upon a sample or a chunk.

for example, PNG should be ran on each sample, whereas lz4 we can run on a chunk.

@benchislett do you think an @property here is the best way to denote this (since we're using classes)?

Reply:
I still think this is out-of-scope. I'd prefer to just do compression sample-wise for now, if that's possible.